src/client-functions/selector-builder/add-api.js

```diff
@@ -0,0 +1,313 @@
+            var node = selector();
+
+            if (!node)
+                return null;
+
+            if (typeof filter === 'string') {
+                return typeof node.querySelectorAll === 'function' ?
+                       node.querySelectorAll(filter) :
+                       null;
+            }
+
+            var result = [];
+
+            var visitNode = currentNode => {
+                var cnLength = currentNode.childNodes.length;
+
+                for (var i = 0; i < cnLength; i++) {
+                    var child = currentNode.childNodes[i];
+
+                    if (filter(child))
+                        result.push(child);
+
+                    visitNode(child);
+                }
+            };
+
+            visitNode(node);
+
+            return result;
+            /* eslint-enable no-undef */
+        }, builderOptions);
+
+        return builder.getFunction();
+    };
+
+    // Parent
+    obj.parent = filter => {
+        if (filter !== void 0)
+            assertFunctionOrStringOnNonNegativeNumber('parent', '"filter" argument', filter);
+
+        var builderOptions = {
+            dependencies: {
+                selector:    getSelector(),
+                filter:      filter,
+                filterNodes: filterNodes
+            }
+        };
+
+        var builder = new SelectorBuilder(() => {
+            /* eslint-disable no-undef */
+            var node = selector();
+
+            if (!node)
+                return null;
+
+            var parents = [];
+
+            for (node = node.parentNode; node; node = node.parentNode)
+                parents.push(node);
+
+            return filter ? filterNodes(parents, filter, document) : parents;
+            /* eslint-enable no-undef */
+        }, builderOptions);
+
+        return builder.getFunction();
+    };
+
+    // Child
+    obj.child = filter => {
+        if (filter !== void 0)
+            assertFunctionOrStringOnNonNegativeNumber('child', '"filter" argument', filter);
+
+        var builderOptions = {
+            dependencies: {
+                selector:    getSelector(),
+                filter:      filter,
+                filterNodes: filterNodes
+            }
+        };
+
+        var builder = new SelectorBuilder(() => {
+            /* eslint-disable no-undef */
+            var node = selector();
+
+            if (!node)
+                return null;
+
+            var childElements = [];
+            var cnLength      = node.childNodes.length;
+
+            for (var i = 0; i < cnLength; i++) {
+                var child = node.childNodes[i];
+
+                if (child.nodeType === 1)
+                    childElements.push(child);
+            }
+
+            return filter ? filterNodes(childElements, filter, node) : childElements;
+            /* eslint-enable no-undef */
+        }, builderOptions);
+
+        return builder.getFunction();
+    };
+
+    // Sibling
+    obj.sibling = filter => {
```

Comment:
I think this is an unusual approach use `sibling`. Why not use `previousSibling` and `nextSibling'?